App Config Provider - Update telemetry (#35556)
Azure/azure-sdk-for-python: sdk/appconfiguration/azure-appconfiguration-provider/README.md

For additional info check out [Dynamic Refresh](https://learn.microsoft.com/azure/azure-app-configuration/enable-dynamic-configuration-python) on MS Learn.

> ## Selecting Feature Flags

Reviewer: This file already contains "Loading Feature Flags" section. Has anything changed?

Author: Not sure how I missed that. Removing this section.

> You can enable loading of feature flags by setting `feature_flags_enabled` to `True`. By default all feature flags with no label are loaded.

Reviewer: > By default all feature flags with no label are loaded.

This statement is a bit confusing. Does that mean we'll always load feature flags with null label by default, even if `feature_flags_enabled` is not set to true?

```python
config = load(
    connection_string=connection_string,
    feature_flag_enabled=True,
)
```

If you want to load feature flags with a specific label you can use `feature_flag_selectors` and provide a list of `SettingSelector`s to filter the feature flags.

```python
config = load(
    connection_string=connection_string,
    feature_flag_enabled=True,
    feature_flag_selectors=[SettingSelector(key_filter="*", label_filter="dev")]
)
```

> Feature flags can be refreshed in a similar way to configurations. By setting `feature_flag_refresh_enabled` to `True` and calling `refresh` on the provider.

Reviewer: ```suggestion
Feature flags can be refreshed in a similar way to configurations. You can set `feature_flag_refresh_enabled` to `True` and call `refresh` on the provider to reload feature flags.
```

```python
config = load(
    connection_string=connection_string,
    feature_flag_enabled=True,
    feature_flag_refresh_enabled=True,
)
```

> 

Reviewer: Can you add the code to trigger refresh of feature flags?

Author: It is there, but in the other section. This one is now removed.
